File: reverse.py
def solve():
    n = int(input())
    a = list(map(int, input().split()))
    j = -1
    ele = None
    for i in range(n):
        if a[i] != (i+1):
            j = i
            ele = (i+1)
            break
    k = -1
    for i in range(j, n):
        if a[i] == ele:
            k = i
            break
    if j != -1:
        for i in range(j):
            print(a[i], end=" ")
        for i in range(k, j-1, -1):
            print(a[i], end=" ")
        for i in range(k+1, n):
            print(a[i], end=" ")
    else:
        for i in range(n):
            print(a[i], end=" ")
    print()

t = int(input())
for _ in range(t):
    solve()

File: reverse.cpp
#include<bits/stdc++.h>
using namespace std;
void solve(){
    int n; cin>>n;
    int a[n]; for(int i=0;i<n;i++) cin>>a[i];
    int j =-1;
    int ele ;
    for(int i=0;i<n;i++){
       if(a[i]!=(i+1)){
          j = i;
          ele = (i+1);
          break;
       }
    }
    int k =-1 ;
    for(int i=j;i<n;i++){
        if(a[i]==ele){
        k = i;
        break;
        }
    }
    if(j!=-1){
    for(int i=0;i<j;i++){
        cout<<a[i]<<" ";
    }
    for(int i=k;i>=j;i--){
        cout<<a[i]<<" ";
    }
    for(int i=k+1;i<n;i++){
        cout<<a[i]<<" ";
    }
    }
    else{
        for(int i=0;i<n;i++){
            cout<<a[i]<<" ";
        }
    }
    cout<<endl;
}
int main(){


int t; cin>>t;
while(t--){
    solve();
}

    return 0;
}